--- /usr/bin/mkdesktophdicons	2003-01-02 00:03:35.000000000 +0100
+++ mkdesktophdicons	2003-04-27 20:12:26.000000000 +0200
@@ -37,7 +37,7 @@
 cat >$HOME/Desktop/"CD-ROM [$d]" <<ENDCDROM
 [Desktop Entry]
 MountPoint=/mnt/auto/$d
-Actions=DVD;Music;Burn;Eject
+Actions=DVD;Music;Burn;Eject;addshare;removeshare
 Dev=$i
 Name=CD-ROM [$d]
 Icon=cdrom_unmount
@@ -82,6 +82,22 @@
 Name[de]=Musik-CD abspielen
 Dev=$i
 MountPoint=/mnt/auto/$d
+
+[Desktop Action addshare]
+Name=Share this CD-ROM
+Name[de]=Dieses CD-ROM freigeben
+Icon=share
+Exec=/usr/sbin/sambastart add "/mnt/auto/$d"
+Dev=$i
+MountPoint=/mnt/auto/$d
+
+[Desktop Action removeshare]
+Name=Remove this share(s) (if shared)
+Name[de]=Diese Freigabe(n) löschen (wenn freigegeben)
+Icon=share
+Exec=/usr/sbin/sambastart remove "/mnt/auto/$d"
+Dev=$i
+MountPoint=/mnt/auto/$d
 ENDCDROM
 rm -f $HOME/.gnome-desktop/$d ; ln -s /mnt/auto/$d $HOME/.gnome-desktop/$d 2>/dev/null
 done
@@ -93,7 +109,7 @@
   d="${p##*/}"
   cat >$HOME/Desktop/"$d" <<ENDHD
 [Desktop Entry]
-Actions=Remount
+Actions=Remount;addshare;removeshare
 Type=FSDevice
 Dev=$p
 MountPoint=$m
@@ -102,6 +118,7 @@
 Icon=hdd_mount
 UnmountIcon=hdd_unmount
 Name=Hard Disk Partition [$d]
+Name[de]=Festplatten-Partition [$d]
 Name[es]=Partición del disco duro [$d]
 
 [Desktop Action Remount]
@@ -110,6 +127,18 @@
 Name=Change read/write mode
 Name[de]=Lese-/Schreibmodus ändern
 Name[es]=Cambiar modo lectura/escritura
+
+[Desktop Action addshare]
+Name=Share this Hard Disk Partition
+Name[de]=Diese Festplatten-Partition freigeben
+Icon=share
+Exec=/usr/sbin/sambastart add "$m"
+
+[Desktop Action removeshare]
+Name=Remove this share(s) (if shared)
+Name[de]=Diese Freigabe(n) löschen (wenn freigegeben)
+Icon=share
+Exec=/usr/sbin/sambastart remove "$m"
 ENDHD
   rm -f $HOME/.gnome-desktop/$d ; ln -s $m $HOME/.gnome-desktop/$d 2>/dev/null
   ;;
